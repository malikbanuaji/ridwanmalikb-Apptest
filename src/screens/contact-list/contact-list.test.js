import * as React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import ContactList from './contact-list';
import {useDispatch, useSelector} from 'react-redux';

jest.mock('./components/contact-item', () => 'ContactItem');
jest.mock('../../features/contact/contactSlice', () => ({
  fetchAllContact: jest.fn(),
  selectAllContact: jest.fn(),
}));

describe('ContactList', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };

  test('should render properly', () => {
    useDispatch.mockImplementationOnce(() => param => param);
    useSelector.mockImplementationOnce(() => [
      {
        firstName: 'BBBB',
        lastName: 'AAAA',
        age: 12,
        photo: 'AA',
        id: '41862780-1627-11ed-a529-d51f5403f685',
      },
    ]);
    const component = render(<ContactList {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render properly onRefresh', () => {
    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);
    useSelector.mockImplementation(() => [
      {
        firstName: 'BBBB',
        lastName: 'AAAA',
        age: 12,
        photo: 'AA',
        id: '41862780-1627-11ed-a529-d51f5403f685',
      },
    ]);
    const component = render(<ContactList {...props} />);
    const list = component.getByTestId('ContactList');
    fireEvent(list, 'refresh');

    expect(dispatch).toBeCalled();
  });

  test('on press contact item', () => {
    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);
    useSelector.mockImplementationOnce(() => [
      {
        firstName: 'BBBB',
        lastName: 'AAAA',
        age: 12,
        photo: 'AA',
        id: '41862780-1627-11ed-a529-d51f5403f685',
      },
    ]);
    const component = render(<ContactList {...props} />);
    const list = component.getByTestId('ContactList');
    const item = list.findByType('ContactItem');
    console.log(item);
    item.props.onPress({id: '1', index: 1});
    expect(props.navigation.navigate).toBeCalled();
  });
});
