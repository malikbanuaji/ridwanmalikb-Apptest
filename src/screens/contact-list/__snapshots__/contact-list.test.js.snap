// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList should render properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#36393f",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 20,
        "paddingTop": 5,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="ContactList"
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
